### PR TITLE
=pro #20495 derive Dist projects from publishArtifact setting

### PR DESCRIPTION
**Pull request by aruediger**
(by removing Dist.distExclude setting)

### CODE REVIEW

**Comment by 2beaucoup** [comment]
refs #20495

**Comment by 2beaucoup** [comment]
`dist` doesn't complete on my machine so I couldn't verify if it actually works.

File: project/Dist.scala [hunk 1]
```diff
@@ -6,62 +6,58 @@ package akka
...
 
   def docsProject: ProjectReference = LocalProject(AkkaBuild.docs.id)
 
-  def aggregated[T](task: SettingKey[Task[T]])(projectRef: ProjectRef, structure: Load.BuildStructure, exclude: Seq[String]): Task[Seq[T]] = {
-    val projects = aggregatedProjects(projectRef, structure, exclude)
-    projects flatMap { task in LocalProject(_) get structure.data } join
-  }
+  def aggregated[T](task: TaskKey[T])(projectRef: ProjectRef, structure: BuildStructure): Task[Seq[T]] =
+    aggregatedProjects(projectRef, structure)
+      .filter { publishArtifact in _ get structure.data getOrElse false }
```
Comment (2beaucoup) [inline project/Dist.scala]: this is the changed filtering

Comment (2beaucoup) [inline project/Dist.scala]: Aggregated projects now get included even if their parents have `publishArtifact := false`. Dunno if that's a problem.

I looked into `ScopeFilter` but only the predefined project filters seems to work. :(

**Comment by johanandren** [comment]
I'll check the PR out and see if dist works on my machine.

**Comment by johanandren** [comment]
This is a list of files ending up in the dist-zip
[dist-contents.txt](https://github.com/akka/akka/files/257587/dist-contents.txt)

**Comment by johanandren** [comment]
Will diff it with dist from master right away

File: project/Dist.scala [hunk 2]
```diff
@@ -6,62 +6,58 @@ package akka
...
 import sbtunidoc.Plugin._
 
 object Dist {
   case class DistSources(depJars: Seq[File], libJars: Seq[File], srcJars: Seq[File], docJars: Seq[File], api: File, docs: File)
 
+  val distDirectory = SettingKey[File]("dist-directory")
+  val distUnzipped = SettingKey[File]("dist-unzipped")
+  val distFile = SettingKey[File]("dist-file")
+
   val distExclude = SettingKey[Seq[String]]("dist-exclude")
```
Comment (patriknw) [inline project/Dist.scala]: is this still used?

**Comment by johanandren** [comment]
Master is 120M, so it seems quite a bunch of things got in there in the branch that isn't in the dist in master
[master.txt](https://github.com/akka/akka/files/257616/master.txt)

File: project/Dist.scala (hunk 2 above)
Comment (2beaucoup) [inline project/Dist.scala]: nope

**Comment by 2beaucoup** [comment]
Thanks for verifying @johanandren! I've pushed an update that should bring the dist result in line with master.

**Comment by johanandren** [comment]
Now the PR branch gives me a 120M dist zip, identical file contents with master.
Here's the listing if anyone else wants to verify: [dist-contents.fixed.txt](https://github.com/akka/akka/files/258682/dist-contents.fixed.txt)

LGTM, great work @2beaucoup !

**Comment by patriknw** [comment]
thanks for verifying @johanandren 
LGTM
great work @2beaucoup

**Comment by 2beaucoup** [comment]
Thanks guys! :+1:

**Pull request merged**